src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -0,0 +1,229 @@
+            username={props.username}
+            profilePicture={props.profilePicture}
+            isFollowing={followType === 'signed'}
+            onPress={() => {
+              handleFollowTopic({
+                type: 'signed'
+              });
+            }}
+          />
+          <UserItem
+            type="incognito"
+            username="Incognito"
+            profilePicture=""
+            isFollowing={followType === 'incognito'}
+            onPress={() => {
+              handleFollowTopic({
+                type: 'incognito'
+              });
+            }}
+          />
+          <View
+            style={{
+              paddingVertical: dimen.normalizeDimen(14),
+              paddingHorizontal: dimen.normalizeDimen(20)
+            }}>
+            <Button
+              onPress={() => {
+                handleUnfollowTopic({
+                  type: followType
+                });
+              }}
+              disabled={followType === ''}
+              styles={{
+                backgroundColor: COLORS.white,
+                paddingVertical: dimen.normalizeDimen(11),
+                paddingHorizontal: dimen.normalizeDimen(20),
+                borderRadius: dimen.normalizeDimen(8),
+                height: dimen.normalizeDimen(40),
+                flexDirection: 'row',
+                justifyContent: 'center',
+                alignItems: 'center',
+                borderColor: isFollow ? COLORS.redalert : COLORS.light_silver,
+                borderWidth: 1
+              }}
+              textStyling={{
+                fontFamily: fonts.inter[500],
+                fontWeight: 'normal',
+                fontSize: 12,
+                color: isFollow ? COLORS.redalert : COLORS.white
+              }}>
+              Leave Community
+            </Button>
+          </View>
+        </View>
+      </BottomSheetV2>
+    </View>
+  );
+});
+
+BottomSheetFollow.displayName = 'BottomSheetFollow';
+
+const styles = StyleSheet.create({
+  containerBottomSheet: {
+    flexDirection: 'column',
+    marginBottom: dimen.normalizeDimen(20)
+  },
+  titleContainer: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    width: '100%',
+    paddingHorizontal: dimen.normalizeDimen(20),
+    paddingVertical: dimen.normalizeDimen(10)
+  },
+  title: {
+    fontFamily: fonts.inter[600],
+    fontWeight: 'bold',
+    fontSize: 16,
+    color: COLORS.black
+  },
+  description: {
+    fontFamily: fonts.inter[400],
+    fontSize: 12,
+    color: COLORS.blackgrey,
+    marginTop: 7
+  },
+  errorText: {
+    fontFamily: fonts.inter[400],
+    fontSize: 12,
+    color: COLORS.redalert,
+    marginTop: dimen.normalizeDimen(7)
+  },
+  button: {
+    marginTop: 33,
+    backgroundColor: COLORS.signed_primary
+  },
+  textStyling: {
+    fontFamily: fonts.inter[600],
+    fontSize: 18,
+    color: COLORS.white
+  }
+});
+
+export default React.memo(BottomSheetFollow);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `BottomSheetFollow` component seems to be well implemented. However, there are a few areas that could be improved for better error handling and maintainability.



<!-- This is an auto-generated comment by OSS CodeRabbit -->